### PR TITLE
Create Subresource Integrity

### PR DESCRIPTION
**Pull request by vabmalikusa**
Subresource Integrity (SRI) is a security feature that enables browsers to verify that  the resources they fetch (for example, from a CDN) are delivered without unexpected  manipulation. It allows you to provide a cryptographic hash that a fetched resource must match.

### CODE REVIEW

File: pages/controls/Subresource Integrity [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+Web sites often rely on resources hosted on third-party servers, such as 
+JavaScript libraries or CSS stylesheets. However, using these resources introduces risks:
+
+1. The third-party server could get compromised, leading to the resources being modified to include malicious content.
+2. Network attacks could modify the resources en route from the third-party server to the user's browser.
+
+SRI helps mitigate these risks by ensuring that the browser only executes resources 
+that match the expected hash. If the resource doesn't match the hash, the browser won't 
+load it, protecting the page from potential attacks.
```
Comment (kingthorin) [inline pages/controls/Subresource Integrity:28]: "...the page and users..." ?

File: pages/controls/Subresource Integrity [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+---
+
+{% include writers.html %}
+
+
+## Description
+
+Subresource Integrity (SRI) is a security feature that enables browsers to verify that 
+the resources they fetch (for example, from a CDN) are delivered without unexpected 
+manipulation. It allows you to provide a cryptographic hash that a fetched resource must match.
```
Comment (kingthorin) [inline pages/controls/Subresource Integrity:18]: This seems directly ripped from: https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity

File: pages/controls/Subresource Integrity [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+- Ensures integrity of externally hosted resources
+- Detects and prevents execution of tampered code
+- Complements other security measures like HTTPS and CSP
+- Helps maintain trust in third-party content
+
+However, SRI also introduces some considerations:
+- Requires knowledge of expected resource hashes
+- Can break sites if resource updates change hashes
+- Doesn't protect against compromise of the main site itself
+- Not all browsers support SRI yet
```
Comment (kingthorin) [inline pages/controls/Subresource Integrity:49]: Seems supported by all current major browsers: https://caniuse.com/?search=integrity

File: pages/controls/Subresource Integrity [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+1. Go to [https://www.srihash.org/](https://www.srihash.org/)
+2. Enter the URL of the resource (in this case, `https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css`)
+3. Click "Generate"
+4. Copy the `integrity` attribute provided
+
+It's important to note that if the resource is updated (for 
+example, if you upgrade to a new version of Bootstrap), you'll 
+need to update the `integrity` hash to match the new version. If you don't, the resource will fail to load.
+
+SRI is a powerful tool for ensuring the integrity of third-party resources,
```
Comment (kingthorin) [inline pages/controls/Subresource Integrity:95]:
```suggestion
SRI is a powerful control for ensuring the integrity of third-party resources,
```

File: pages/controls/Subresource Integrity [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+ but it's not a complete security solution on its own. It should be used 
+ in combination with other security best practices, like serving your site over 
+ HTTPS and using Content Security Policy (CSP).
+
+## References
+- [MDN Web Docs: Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
```
Comment (kingthorin) [inline pages/controls/Subresource Integrity:101]:
```suggestion
## References

- [MDN Web Docs: Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
```

File: pages/controls/Subresource Integrity (hunk 2 above)
Comment (vabmalikusa) [reply pages/controls/Subresource Integrity:18]: Sure. Adding in references. Also changed

File: pages/controls/Subresource Integrity (hunk 5 above)
Comment (vabmalikusa) [reply pages/controls/Subresource Integrity]: Done

File: pages/controls/Subresource Integrity (hunk 4 above)
Comment (vabmalikusa) [reply pages/controls/Subresource Integrity]: Done

File: pages/controls/Subresource Integrity (hunk 3 above)
Comment (vabmalikusa) [reply pages/controls/Subresource Integrity]: Ok

File: pages/controls/Subresource Integrity (hunk 1 above)
Comment (vabmalikusa) [reply pages/controls/Subresource Integrity]: Added

**Review by kingthorin — APPROVED** [review]
Thanks

**Pull request merged**